recipes/mongo-c-driver/all/conanfile.py

```diff
@@ -0,0 +1,226 @@
+        if self.options.with_sasl == "sspi" and self.settings.os != "Windows":
+            raise ConanInvalidConfiguration("with_sasl=sspi only allowed on Windows")
+
+    def requirements(self):
+        if self.options.with_ssl == "openssl":
+            self.requires("openssl/1.1.1h")
+        elif self.options.with_ssl == "libressl":
+            self.requires("libressl/3.2.0")
+        if self.options.with_sasl == "cyrus":
+            self.requires("cyrus-sasl/2.1.27")
+        if self.options.with_snappy:
+            self.requires("snappy/1.1.8")
+        if self.options.with_zlib:
+            self.requires("zlib/1.2.11")
+        if self.options.with_zstd:
+            self.requires("zstd/1.4.5")
+        if self.options.with_icu:
+            self.requires("icu/68.1")
+
+    def build_requirements(self):
+        if self.options.with_ssl == "libressl" or self.options.with_zstd:
+            self.build_requires("pkgconf/1.7.3")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        # Fix Snappy
+        tools.replace_in_file(os.path.join(self._source_subfolder, "src", "libmongoc", "CMakeLists.txt"),
+                              "include (FindSnappy)\nif (SNAPPY_INCLUDE_DIRS)",
+                              "if(ENABLE_SNAPPY MATCHES \"ON\")\n  find_package(Snappy REQUIRED)")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "src", "libmongoc", "CMakeLists.txt"),
+                              "SNAPPY_LIBRARIES",
+                              "Snappy_LIBRARIES")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "src", "libmongoc", "CMakeLists.txt"),
+                              "SNAPPY_INCLUDE_DIRS",
+                              "Snappy_INCLUDE_DIRS")
+        # Fix Openssl
+        tools.replace_in_file(os.path.join(self._source_subfolder, "src", "libmongoc", "CMakeLists.txt"),
+                              "OPENSSL_FOUND", "OpenSSL_FOUND")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "src", "libmongoc", "CMakeLists.txt"),
+                              "OPENSSL_VERSION", "OpenSSL_VERSION")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "src", "libmongoc", "CMakeLists.txt"),
+                              "OPENSSL_CRYPTO_LIBRARY", "OpenSSL_Crypto_LIBS")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "src", "libmongoc", "CMakeLists.txt"),
+                              "OPENSSL_LIBRARIES", "OpenSSL_LIBRARIES")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "src", "libmongoc", "CMakeLists.txt"),
+                              "OPENSSL_INCLUDE_DIR", "OpenSSL_INCLUDE_DIR")
```

Comment:
```suggestion
        to_replace_old_new = [
            # Fix Snappy
            {"old": "include (FindSnappy)\nif (SNAPPY_INCLUDE_DIRS)",
             "new": "if(ENABLE_SNAPPY MATCHES \"ON\")\n  find_package(Snappy REQUIRED)"},
            {"old": "SNAPPY_LIBRARIES", "new": "Snappy_LIBRARIES"},
            {"old": "SNAPPY_INCLUDE_DIRS", "new": "Snappy_LIBRARIES"},
            # Fix Openssl
            {"old": "OPENSSL_FOUND", "new": "OpenSSL_FOUND"},
            {"old": "OPENSSL_VERSION", "new": "OpenSSL_VERSION"},
            {"old": "OPENSSL_CRYPTO_LIBRARY", "new": "OpenSSL_Crypto_LIBS"},
            {"old": "OPENSSL_LIBRARIES", "new": "OpenSSL_LIBRARIES"},
            {"old": "OPENSSL_INCLUDE_DIR", "new": "OpenSSL_INCLUDE_DIR"},
            # Fix LibreSSL
            {"old": "set (SSL_LIBRARIES -ltls -lcrypto)", "new": ""},
        ]
        for old_new in to_replace_old_new:
            tools.replace_in_file(os.path.join(self._source_subfolder, "src", "libmongoc", "CMakeLists.txt"),
                                  old_new["old"], old_new["new"])
```

Last modification. I've checked libressl build with #2933. `set (SSL_LIBRARIES -ltls -lcrypto)` has to be removed in mongo-c-driver CMakeLists.txt